validator/client/beacon-api/genesis.go

```diff
@@ -0,0 +1,71 @@
+//go:build use_beacon_api
+// +build use_beacon_api
+
+package beacon_api
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/api/gateway/apimiddleware"
+	rpcmiddleware "github.com/prysmaticlabs/prysm/v3/beacon-chain/rpc/apimiddleware"
+	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+)
+
+func (c beaconApiValidatorClient) waitForChainStart() (*ethpb.ChainStartResponse, error) {
+	genesis, err := c.getGenesis()
+	if err != nil {
+		return nil, err
+	}
+
+	genesisTime, err := strconv.ParseUint(genesis.Data.GenesisTime, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	chainStartResponse := &ethpb.ChainStartResponse{}
+	chainStartResponse.Started = true
+	chainStartResponse.GenesisTime = genesisTime
+
+	// Remove the leading 0x from the string before decoding it to bytes
+	genesisValidatorRoot, err := hex.DecodeString(genesis.Data.GenesisValidatorsRoot[2:])
+	if err != nil {
+		return nil, err
+	}
+	chainStartResponse.GenesisValidatorsRoot = genesisValidatorRoot
+
+	return chainStartResponse, nil
+}
+
+func (c beaconApiValidatorClient) getGenesis() (*rpcmiddleware.GenesisResponseJson, error) {
+	resp, err := c.httpClient.Get(c.url + "/eth/v1/beacon/genesis")
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err = resp.Body.Close(); err != nil {
+			return
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		errorJson := apimiddleware.DefaultErrorJson{}
+		err = json.NewDecoder(resp.Body).Decode(&errorJson)
```

Comment:
This won't work with `DefaultErrorJson` because it has a `Code` field, not `StatusCode`. Looking back on it, I think that having `StatusCode` will make life easier. Can you open a separate PR that changes the field name (and the associated getter method which is now called `StatusCode`)? We can merge that PR first and then the problem will go away.

Reply:
Why won't this work? The Beacon API returns this:

```json
{
  "code": 404,
  "message": "Chain genesis info is not yet known"
}
```

And `DefaultErrorJson` is serialized the same way:

```go
type DefaultErrorJson struct {
    Message string `json:"message"`
    Code    int    `json:"code"`
}
```

`resp.StatusCode` here is only the status of the http request, not the name of the field in the json schema.

Reply:
Thinking of it a little bit more, I doubt either option will work. This is because the body does not contain fields `message` and `code`. In my opinion you should write:
```
if resp.StatusCode != http.StatusOK {
	errorJson := apimiddleware.DefaultErrorJson{Message: string(resp.Body), Code: resp.StatusCode}